src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -397,8 +402,69 @@ def makeSubscriptions(self, workflow):
             pass
 
         elif self.msConfig['defaultDataManSys'] == 'Rucio':
-            pass
+            if isinstance(workflow, MSOutputTemplate):
+                msg = "Making transfer subscriptions for %s"
+                self.logger.info(msg, workflow['RequestName'])
 
+                rucioResultList = []
+                if workflow['destinationOutputMap']:
+                    for dMap in workflow['destinationOutputMap']:
+                        try:
+                            msg = "Making transfer subscriptions for %s"
+                            self.logger.info(msg, workflow['RequestName'])
+                            copies = workflow['numberOfCopies'] if workflow['numberOfCopies'] else 1
+                            # NOTE:
+                            #    Once we get rid of DDM this rseExpression generation
+                            #    should go in the Producer thread
+                            if workflow['isRelVal']:
+                                pnnExpr = '('+'|'.join(["pnn=%s" % pnn for pnn in dMap['destination']])+')'
+                                rseExpression = pnnExpr + '&cms_type=real&rse_type=DISK'
+                            else:
+                                rseExpression = '(tier=2|tier=1)&cms_type=real&rse_type=DISK'
+                            scope = 'cms'
+                            if self.msConfig['enableDataPlacement']:
+                                rucioResultList.append(self.rucio.createReplicationRule(dMap['datasets'],
+                                                                                        rseExpression,
+                                                                                        copies=copies,
+                                                                                        scope=scope))
+                            else:
+                                msg = "Dry run mode. The effective Rucio submission would look like: \n"
+                                msg += "account: %s \n"
+                                msg += "dids: %s \n"
+                                msg += "rseExpression: %s\n"
+                                msg += "copies: %s\n"
+                                self.logger.warning(msg,
+                                                    self.msConfig['rucioAccount'],
+                                                    pformat([{'scope': scope, 'name': did} for did in dMap['datasets']]),
```

Comment:
this scope is also not needed here. I don't think production will ever use anything but `cms`

Reply:
I think it doesn't hurt to see the transfer submission, printed exactly the same way as it would be sent to Rucio if the run mode was `REAL`. That's why I put it here and above too.

Reply:
I actually wanted to save those cpu cycles spent in the for loop. So one could something like:
```
msg += "scope: %s \n"
msg += "dids: %s \n" etc etc
```

Reply:
I think this can be updated as well (then check if pformat still needs to be imported)

Reply:
Done. But I kept `pformat` - otherwise the message get completely unstructured.